pkgs/build-support/build-fhs-userenv/default.nix

```diff
@@ -0,0 +1,37 @@
+{ writeTextFile, stdenv, ruby } : { env, runScript } :
+
+let
+  name = env.pname;
+
+  # Sandboxing script
+  chroot-user = writeTextFile {
+    name = "chroot-user";
+    executable = true;
+    destination = "/bin/chroot-user";
+    text = ''
+      #! ${ruby}/bin/ruby
+      ${builtins.readFile ./chroot-user.rb}
```

Comment:
If I understood correctly, the problem was because of user reading a big file in his configuration on evaluation and writing it to the output. Because our `writeTextFile` is implemented via `echo -n`, this resulted in `Argument list too long` error/ However, here I read a script from `nixpkgs`, so we (as maintainers) control the length of file, and if it gets too large after some change we'll invent another way to write it before it would break anything for users.

Another way of fixing this would be to change `writeTextFile` to output the file to the nix store first via `toFile` builtin, and then use `cat` in its build script -- then we would have no restriction on `text` size.